### PR TITLE
[`flake8-pyi`] Implement PYI053

### PR DESCRIPTION
**Pull request by density**
## Summary

Implement Y053 from https://github.com/PyCQA/flake8-pyi: `Only string and bytes literals <=50 characters long are permitted.`

## Test Plan

Added snapshot tests

ref: #848

### CODE REVIEW

File: crates/ruff/src/rules/flake8_pyi/rules/long_string_or_bytes_in_stub.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+#[violation]
+pub struct LongStringOrBytesInStub;
+
+/// ## What it does
+/// Checks for `str` or `bytes` literals longer than 50 characters in stubs.
+///
+/// ## Why is this bad?
+/// Extremely long strings used as argument defaults or otherwise are unlikely to be useful for
+/// users. Consider replacing them with ellipses.
```
Comment (density) [inline crates/ruff/src/rules/flake8_pyi/rules/long_string_or_bytes_in_stub.rs:16]: Questionable reasoning, but my archaeology in `flake8-pyi` didn't turn up any better reasons.

File: crates/ruff/src/rules/flake8_pyi/rules/long_string_or_bytes_in_stub.rs
Comment (density) [inline crates/ruff/src/rules/flake8_pyi/rules/long_string_or_bytes_in_stub.rs:1]: `_in_stub` feels redundant in the name given that it's in `flake8_pyi`. lmk if you prefer a different name.

File: crates/ruff/src/rules/flake8_pyi/rules/long_string_or_bytes_in_stub.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/ruff/src/rules/flake8_pyi/rules/long_string_or_bytes_in_stub.rs:16]: Here's an explanation: https://github.com/PyCQA/flake8-pyi/blob/ceab86d16b822d12abae1d8087850d0bc66d2f52/CHANGELOG.md?plain=1#LL110C1-L115C23

Comment (density) [reply crates/ruff/src/rules/flake8_pyi/rules/long_string_or_bytes_in_stub.rs:16]: Funny, good catch.

**Review by charliermarsh — COMMENTED** [review]
Do you mind taking a look at the `is_valid_default_value_with_annotation` logic in `rules/flake8_pyi/rules/simple_defaults.rs`? It looks like these checks _used_ to be part of Y015, but were made more general and moved to their own rule (Y053) in flake8-pyi -- you can see the commit [here](https://github.com/PyCQA/flake8-pyi/commit/f6118974eed9a2feea170d7634c32b1eaa138316).

So when adding this, we should probably change the `is_valid_default_value_with_annotation` definition to ignore overlong strings (since they'll now be picked up by this new rule).

**Comment by density** [comment]
@charliermarsh Thanks for the feedback, updated.

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff/src/rules/flake8_pyi/rules/long_string_or_bytes_in_stub.rs (hunk 1 above)
Comment (density) [reply crates/ruff/src/rules/flake8_pyi/rules/long_string_or_bytes_in_stub.rs]: @charliermarsh I improved the comment and added an example.

**Pull request merged**